libbeat/logp/core_test.go

```diff
@@ -167,6 +167,8 @@ func TestLoggerLevel(t *testing.T) {
 }
 
 func TestRecover(t *testing.T) {
+	t.Skip("go.uber.org/zap does not support Golang 1.12. See: https://github.com/uber-go/zap/issues/682")
+
```

Comment:
@kvch What's up with this? I am trying to understand the impact of the above issue on us if we are upgrading. My questions are:

- Did you get the same errors as the issue above?
- I currently don't see any tracking on the zap repo.
- Looking at the golang issue some commits where done to address the problem.

This look like an issue concerning the extracted line numbers when a panic occurs.

Reply:
The caller of the logging function is not returned correctly. Their unit tests also catch this problem, as shown in the issue I have linked.

This is the error I am getting when running the test:
```
--- FAIL: TestRecover (0.00s)
    core_test.go:180:
                Error Trace:    core_test.go:180
                                                        core_test.go:188
                Error:          Not equal:
                                expected: "logp/core_test.go"
                                actual  : "testing/testing.go"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -logp/core_test.go
                                +testing/testing.go
                Test:           TestRecover
FAIL
exit status 1
FAIL    github.com/elastic/beats/libbeat/logp   0.003s
```
The caller is incorrect.

I have tried first to see if updating the package helps. Unfortunately, it does not.

I have doubts whether it is going to be fixed soon. I would fork a repo and fix it. Until the fix is added to upstream, I would use Beats with the fork. WDYT?
I think Golang 1.12 brings a lot of value which is worth the additional effort.

Reply:
It's not a bug. I had to do a minor follow-up in `logp/global.go` to decrease the number of caller skipped, as the panic tracebacks had been changed. (https://go-review.googlesource.com/c/go/+/153477/)